src/generic.rs

```diff
@@ -0,0 +1,24 @@
+use rug::{Integer,rand::RandState};
+///Init public key structure for elgamal encryption.
```

Comment:
this is actually not generic anymore, consider how do you move this code or use generic type.